be/test/vec/columns/common_column_test.h

```diff
@@ -0,0 +1,766 @@
+    // like vector column, the resize will resize the data column
+    // like array column, the resize will resize the offsets column and the data column (which in creator we have check staff for the size of data column is the same as the size of offsets column)
+    void resizeAssert(MutableColumnPtr col, size_t expect_size) {
+        col->resize(expect_size);
+        EXPECT_EQ(col->size(), expect_size);
+    }
+
+    // replicate is clone with new column from the origin column, always from ColumnConst to expand the column
+    void replicateAssert(MutableColumnPtr col, IColumn::Offsets& offsets) {
+        auto new_col = col->replicate(offsets);
+        EXPECT_EQ(new_col->size(), offsets.back());
+    }
+
+    // byte size is just appriximate size of the column
+    //  as fixed column type, like column_vector, the byte size is sizeof(columnType) * size()
+    //  as variable column type, like column_string, the byte size is sum of chars size() and offsets size * sizeof(offsetType)
+    void byteSizeAssert(MutableColumnPtr col, size_t expect_size) {
+        EXPECT_EQ(col->byte_size(), expect_size);
+    }
+
+    // allocated bytes is the real size of the column
+    void allocatedBytesAssert(MutableColumnPtr col, size_t expect_size) {
+        EXPECT_EQ(col->allocated_bytes(), expect_size);
+    }
+
+    // clone_resized will clone the column and cut/append to the new column with the size of the original column
+    void cloneResizedAssert(MutableColumnPtr col, size_t expect_size) {
+        auto new_col = col->clone_resized(expect_size);
+        EXPECT_EQ(new_col->size(), expect_size);
+    }
+
+    // get_shrinked_column should only happened in char-type column or nested char-type column
+    //  just shrink the end zeros for char-type column which happened in segmentIterator
+    //    eg. column_desc: char(6), insert into char(3), the char(3) will padding the 3 zeros at the end for writing to disk.
+    //       but we select should just print the char(3) without the padding zeros
+    //  limit and topN operation will trigger this function call
+    void getShrinkedColumnAssert(MutableColumnPtr col, size_t spcific_size_defined) {
+        EXPECT_TRUE(col->could_shrinked_column());
+        auto new_col = col->get_shrinked_column();
+        for (size_t i = 0; i < new_col->size(); i++) {
+            EXPECT_EQ(col->get_data_at(i).size, spcific_size_defined);
+        }
+    }
+
+    //serialize and deserialize which usually used in AGG function:
+    //  serialize_value_into_arena, deserialize_and_insert_from_arena (called by AggregateFunctionDistinctMultipleGenericData, group_array_intersect, nested-types serder like: DataTypeArraySerDe::write_one_cell_to_jsonb)
+    void ser_deserialize_with_arena_impl(MutableColumns& columns, const DataTypes& data_types) {
```

Comment:
why test many columns at one time.

Reply:
because this situation is can pass some columns at one time , which in serialize_value_into_arena/deserialize_and_insert_from_arena caller